src/vs/platform/terminal/common/terminal.ts

```diff
@@ -167,6 +167,8 @@ export interface IPtyHostAttachTarget {
 	icon: TerminalIcon | undefined;
 	fixedDimensions: IFixedTerminalDimensions | undefined;
 	environmentVariableCollections: ISerializableEnvironmentVariableCollections | undefined;
+	reconnectionOwner?: string;
+	task?: { label: string; id: string; lastTask?: string; group?: string };
```

Comment:
Can this be a reconnection owner id or something and this data gets persisted by contrib/tasks? We don't want to mention tasks anywhere in terminal ideally

Reply:
This?

```json
reconnectionOwner: {
  ownerName: string;
  ownerId: number;
}
```

Reply:
If that's not easily doable, a temporary fix is to give this a generic name with type `unknown` and validate the type in tasks. I'd be fine for that to live in terminal as it basically moves the tech debt out of terminal and into tasks.